src/include/wiredtiger.in

```diff
@@ -5408,6 +5443,167 @@ struct __wt_storage_source {
+     * @param session the current WiredTiger session
+     * @param checkpoint_id the checkpoint id returned
+     */
+    int (*pl_get_complete_checkpoint)(WT_PAGE_LOG *page_log, WT_SESSION *session, uint64_t *checkpoint_id);
+
+    /*!
+     * Decode a portion of the package. This can be called successively with a returned package
+     * buffer to get the full page image, when package_id is zero, and any attached deltas.
+     * Returns WT_NOTFOUND when the delta_number is too large, that is, the set of returns
+     * has been exhausted.
+     *
+     * @errors
+     *
+     * @param page_log the WT_PAGE_LOG
+     * @param session the current WiredTiger session
+     * @param package_buffer the buffer returned by WT_PAGE_LOG_HANDLE::plh_get.
+     * @param delta_number the index of the delta number.  Using a delta number of 0 indicates
+     *        that the full page image is returned.  The full page image is always available.
+     *        Otherwise, the index number has no meaning except as a way to step through the
+     *        ordered list of returned deltas.
+     */
+    int (*pl_get_package_part)(WT_PAGE_LOG *page_log, WT_SESSION *session, WT_ITEM *package_buffer,
+         int delta_number, WT_ITEM *result);
+
+    /*!
+     * Open a handle for further operations on a table.
+     *
+     * @errors
+     *
+     * @param page_log the WT_PAGE_LOG
+     * @param session the current WiredTiger session
+     * @param table_id the unique table id for the given table
+     * @param plh the returned handle
+     */
+    int (*pl_open_handle)(WT_PAGE_LOG *page_log, WT_SESSION *session, uint64_t table_id, WT_PAGE_LOG_HANDLE **plh);
+
+    /*!
+     * A callback performed when the log page service or reference is closed
+     * and will no longer be used.  The initial creation of the log page service
+     * counts as a reference, and each call to WT_PAGE_LOG::add_reference
+     * increase the number of references.  When all references are released, the
+     * log page service and any resources associated with it are released.
+     *
+     * This method is not required and should be set to NULL when not
+     * required by the log page service implementation.
+     *
+     * The WT_PAGE_LOG::terminate callback is intended to allow cleanup;
+     * the handle will not be subsequently accessed by WiredTiger.
+     */
+    int (*terminate)(WT_PAGE_LOG *page_log, WT_SESSION *session);
+};
+
+struct __wt_page_log_handle {
+    /*!
+     * Put an object into the paging/logging service.
+     *
+     * @errors
+     *
+     * @param plh the WT_PAGE_LOG_HANDLE
+     * @param session the current WiredTiger session
+     * @param page_id the page to be written
+     * @param checkpoint_id the checkpoint to be written to
+     * @param is_delta is true if this is a delta, otherwise it is a full page image
+     * @param buf content to be written
+     */
+    int (*plh_put)(WT_PAGE_LOG_HANDLE *plh, WT_SESSION *session,
```

Comment:
Can you please ensure that this method and the get method below follow the expected API specs?

For example, based on the Page Service deep dive notes, and based on the corresponding protobuf specs, we need to also take as an input the base checkpoint ID and a backlink checkpoint ID. Similarly, we need to get that metadata during get. There may be more, such as a number of deltas since the last full image; not sure what the protobuf says off the top of my head.

Reply:
Yes, I will fix this in my next PR - I will merge it the way it is now so @korteland and @quchenhao can start some experiments.